Automattic/jetpack
# Adds a new Photon use mode that disables intermediate image sizes.

#### Changes proposed in this Pull Request:
Adds a new filter that when used will enable Photon noresize mode:
* effectively disabling intermediate image size creation;
* loading all images from Photon on the client size, as always;
* loading all media from Photon on the wp-admin side as well, with exceptions;

#### Testing instructions:
This is still a work in progress, please report any problems that you find with the points below, any help is hugely appreciated.

* Enable the noresize mode using the filter:
```
add_filter( 'jetpack_photon_noresize_mode', '__return_true' );
```
* Make sure your images still work: Media manager, existing images, existing images in posts.
* Make sure you can upload and see your media.
* Make sure you can crop and resize existing images.
* Make sure both classic and Gutenberg editors work with images as they should.
* Use third party plugins that work with images, there should be no conflicts.

#### Proposed changelog entry for your changes:
* New mode added to Photon that allows to save disk space by disabling resized image creation.
